thalo-eventstoredb/src/event_store.rs

```diff
@@ -0,0 +1,340 @@
+use std::{fmt::Debug, vec};
+
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use eventstore::{
+    AppendToStreamOptions, Client, EventData, ExpectedRevision, ReadAllOptions, ReadStreamOptions,
+    StreamPosition, ResolvedEvent,
+};
+use futures::TryFutureExt;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use thalo::{
+    aggregate::{Aggregate, TypeId},
+    event::{AggregateEventEnvelope, EventType},
+    event_store::EventStore,
+};
+use uuid::Uuid;
+
+use crate::Error;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ESDBEventPayload {
+    created_at: DateTime<Utc>,
+    aggregate_type: String,
+    aggregate_id: String,
+    event_data: serde_json::Value,
+}
+
+impl ESDBEventPayload {
+    pub fn event_envelope<A>(&self, id: usize) -> Result<AggregateEventEnvelope<A>, Error>
+    where
+        A: Aggregate,
+        <A as Aggregate>::Event: DeserializeOwned,
+    {
+        Ok(AggregateEventEnvelope::<A> {
+            id,
+            created_at: self.created_at.into(),
+            aggregate_type: self.aggregate_type.to_string(),
+            aggregate_id: self.aggregate_id.clone(),
+            sequence: id,
+            event: serde_json::from_value(self.event_data.clone())
+                .map_err(Error::DeserializeEvent)?,
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct ESDBEventStore {
+    pub client: Client,
```

Comment:
Do users typically need to access the client directly?
If so, could we make this private and have a method `fn client(&self) -> &Client` to access the client. This allows us to modify the internal structure without making breaking changes.

Reply:
Nope - I think this was a temp change to make it easier to debug when I plugged it into the /examples. Good catch!